src/app/shared/pipes/groupPermissionCaption.ts

```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+const PERMISSION_CAPTIONS = {
+  none: $localize`None`,
+  children: $localize`Children`,
+  all: $localize`All`,
+  all_with_grant: $localize`All with grant`,
+  enter: $localize`Enter`,
+  content: $localize`Content`,
+  content_with_descendants: $localize`Content with descendants`,
+  solution: $localize`Solution`,
+  solution_with_grant: $localize`Solution with grant`,
+  info: $localize`Info`,
+  result: $localize`Result`,
+  answer: $localize`Answer`,
+  answer_with_grant: $localize`Answer with grant`,
+};
+
+@Pipe({ name: 'groupPermissionCaption', pure: true })
+export class GroupPermissionCaptionPipe implements PipeTransform {
+  constructor() {}
+
+  transform(value: string): string {
+    return new Map<string, string>(Object.entries(PERMISSION_CAPTIONS)).get(value) || $localize`No caption`;
```

Comment:
Why not just :arrow_down: ?
```js
PERMISSION_CAPTIONS[value] ?? $localize`No caption`
```

Reply:
I got error: TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type

Reply:
Fixed